### PR TITLE
[MC-602] Create Enrich InsightIDR From VirusTotal to Slack workflow

### PR DESCRIPTION
**Pull request by mgrandys-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

<details>

```
[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Executing validator WorkflowEncodingValidator
[*] Executing validator WorkflowParametersKeywordValidator
[*] Executing validator WorkflowUseCaseValidator
[*] Executing validator UnapprovedKeywordsValidator
[*] Executing validator WorkflowPythonScriptUseValidator

----
[*] Total time elapsed: 1913.199ms
```

<summary>
icon-validate --all .
</summary>
</details>

### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

Workflow

Workflow_Domains_Loop

Workflow_Hashes_Loop

Workflow_IPs_Loop

Workflow_Processes_Loop

Workflow_URLs_Loop

#### Job
Job (1)

#### Artifact

Domain_Artifact

Hash_Artifact

IP_Artifact

URL_Artifact

#### Slack

Slack_Domain_Output

Slack_Hash_Output

Slack_IP_Output

Slack_URL_Output

### CODE REVIEW

**Comment by mgrandys-r7** [comment]
Jobs:
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/3c3db52b-7a6f-4b93-96a1-e6682c6af7bd
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/e06cc1c4-8623-4a01-a44b-fb2600097fc1
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/7a8c2a44-ed1f-4915-acd6-1c8774b6e766
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/2345c344-da86-4c4a-96e6-78e00acb1dfe

**Comment by mgrandys-r7** [comment]
Updated workflow and jobs.

Workflow:
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/workflows/details/3b1b9a46-5530-4882-9bad-38577a9da25e/versions/db4b1d44-3907-4fda-b9d3-05c9bc525170

Jobs:
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/3c3c62d5-a139-4a63-b743-192bee137855
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/cb760650-e429-4331-85a8-635c422f32ca
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/36942a2e-402a-4ae3-978e-064348598233
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/ccfcb6bc-d18e-4791-a344-391e412b8369

**Comment by dsliwinski-r7** [comment]
@mgrandys-r7 I think the "Up Next" section can be removed from artifacts and messages. In the case of reports for IP addresses, the link to the complete report from VirusTotal is incorrect. Could you please make sure these links are working?

**Comment by mgrandys-r7** [comment]
@dsliwinski-r7 You are right, I will make those changes after @r7-kszczepanskagorna review

**Review by r7-kszczepanskagorna — DISMISSED** [review]

**Comment by mgrandys-r7** [comment]
After PR:

Workflow:
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/workflows/details/3b1b9a46-5530-4882-9bad-38577a9da25e/versions/d44b7bbf-7ede-4ed0-a7fd-f2222e70daad

Jobs:
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/a8a86a45-5ac0-4b08-a5a4-fc7c985b5bea
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/5c0aed67-e1fc-45fc-9300-c77a674c3dbf
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/cc90c83f-0e8e-4110-a4c1-0a0791b8505d
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/dc991df4-81a9-4015-814c-fa749f78800c

**Comment by mgrandys-r7** [comment]
> @mgrandys-r7 I think the "Up Next" section can be removed from artifacts and messages. In the case of reports for IP addresses, the link to the complete report from VirusTotal is incorrect. Could you please make sure these links are working?

Done

**Review by dsliwinski-r7 — DISMISSED** [review]

**Review by r7-kszczepanskagorna — DISMISSED** [review]

**Comment by mgrandys-r7** [comment]
In previous version validator passed, but based on this message I made changes in workflow name:
https://rapid7.slack.com/archives/G012ZKN6PCZ/p1650902675506859?thread_ts=1650898853.155939&cid=G012ZKN6PCZ

File: workflows/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack.icon [hunk 1]
```diff
@@ -0,0 +1,5625 @@
...
+            "type": "action",
+            "plugin": {
+              "name": "VirusTotal",
+              "slugVendor": "rapid7",
+              "slugName": "virustotal",
+              "slugVersion": "8.1.0",
+              "imageData": "https://us.cdn-assets.connect.insight.rapid7.com/icons/rapid7/virustotal/8.1.0/icon.png"
+            },
+            "identifier": "lookup_hash",
+            "continueOnFailure": true,
```
Comment (ablakley-r7) [inline workflows/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack.icon:1158]: Workflow looks great but wanted to check if 'continueOnFailure' should be set false for these actions?

Comment (mgrandys-r7) [reply workflows/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack.icon:1158]: It's fine. As this workflow is iterating through a loop of domains/hashes etc we don't the whole workflow to fail if a single value is invalid.

**Comment by mgrandys-r7** [comment]
Major updated, workflow tested

Workflow:
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/workflows/details/5e53e094-7584-436e-9255-8fe207238986/versions/2019524a-30f2-40d5-8f24-cdb9e376ef2d

Jobs:
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/cbbdcd0d-7a31-4a4b-bbb7-4e69e251e283
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/e0236e07-cb3a-4e18-ad96-7fa9a68a9694
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/a04aef53-c218-4f56-9ca4-1ca1c44cc26e
https://us.connect.insight.rapid7.com/99AD7AE6317E5DAEB626#/jobs/details/7e0fe2bc-738b-4460-83b3-ce7e847f99f8

File: workflows/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack/workflow.spec.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+extension: workflow
+products: ["insightconnect", "insightidr"]
+name: Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack
+title: "Enrich InsightIDR Alerts With Threat Intelligence From VirusTotal To Slack"
```
Comment (ablakley-r7) [inline workflows/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack/workflow.spec.yaml:4]: I believe if this is changed to 'with' and rebuilt it should fix the build failure
```suggestion
title: "Enrich InsightIDR Alerts with Threat Intelligence From VirusTotal To Slack"
```

Comment (mgrandys-r7) [reply workflows/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack/workflow.spec.yaml:4]: @ablakley-r7 If I change to `with` then validator is throwing issue that `From` should be also lowercase. Changing to `from` makes validator to pass - this naming version was released and approved 13th April but as I mentioned above I was asked by @blaxminarayan-r7 here: https://rapid7.slack.com/archives/G012ZKN6PCZ/p1650902675506859?thread_ts=1650898853.155939&cid=G012ZKN6PCZ to change it so now validator is failing.

Comment (ablakley-r7) [reply workflows/Enrich_InsightIDR_Alerts_With_Threat_Intelligence_From_VirusTotal_To_Slack/workflow.spec.yaml:4]: Sorry completely missed that comment, all looks great then!

**Pull request merged**